### PR TITLE
multiapp example with controlled dirac point loads on sub-app.

### PR DESCRIPTION
**Pull request by lynnmunday**
Still need to check that they change with each optimization iteration

### CODE REVIEW

**Comment by lynnmunday** [comment]
This was a test to see if I could request you all.  I already merged in the pr for the transfer code myself.

File: test/tests/multiapps/optimiation_transfer/master.i [hunk 1]
```diff
@@ -82,8 +82,10 @@
   []
   [opt_results]
     type = OptimizationVectorPostprocessor
-    parameters = 'BCs/left/value BCs/right/value BCs/top/value'
-    intial_values = '1 2 3'
+    parameters = 'DiracKernels/pt0/value
+    DiracKernels/pt1/value
+    DiracKernels/pt2/value'
+    intial_values = '100 200 300'
```
Comment (dewenyushu) [inline test/tests/multiapps/optimiation_transfer/master.i:88]: I see here you are switching to adding point loads using controllable variables for the forward problem. In theory (future work) we should have another block for setting the loads for the adjoint problem, right?

**Review by dewenyushu — APPROVED** [review]
Let's keep doing this! :)

File: test/tests/multiapps/optimiation_transfer/master.i (hunk 1 above)
Comment (lynnmunday) [reply test/tests/multiapps/optimiation_transfer/master.i:88]: I think we would have a separate multiapp, transfer and vpp for the adjoint problem.  I think the current vpp and transfer should be everything we need for the adjoint.  I think we need a new multiapp that can use the execution flags that zach made to either run the forward problem or to run the forward and adjiont problem for the gradient solve.

**Pull request merged**